papers/language_specification/Marlowe.tex

```diff
@@ -0,0 +1,183 @@
+ & {\delimit}  &{\terminal{TimeIntervalStart}}  \\
+ & {\delimit}  &{\terminal{TimeIntervalEnd}}  \\
+ & {\delimit}  &{\terminal{UseValue}} {\nonterminal{ValueId}}  \\
+ & {\delimit}  &{\terminal{Cond}} {\nonterminal{Observation}} {\nonterminal{Value}} {\nonterminal{Value}}  \\
+ & {\delimit}  &{\terminal{(}} {\nonterminal{Value}} {\terminal{)}}  \\
+\end{tabular}\\
+
+\begin{tabular}{lll}
+{\nonterminal{Observation}} & {\arrow}  &{\terminal{AndObs}} {\nonterminal{Observation}} {\nonterminal{Observation}}  \\
+ & {\delimit}  &{\terminal{OrObs}} {\nonterminal{Observation}} {\nonterminal{Observation}}  \\
+ & {\delimit}  &{\terminal{NotObs}} {\nonterminal{Observation}}  \\
+ & {\delimit}  &{\terminal{ChoseSomething}} {\nonterminal{ChoiceId}}  \\
+ & {\delimit}  &{\terminal{ValueGE}} {\nonterminal{Value}} {\nonterminal{Value}}  \\
+ & {\delimit}  &{\terminal{ValueGT}} {\nonterminal{Value}} {\nonterminal{Value}}  \\
+ & {\delimit}  &{\terminal{ValueLT}} {\nonterminal{Value}} {\nonterminal{Value}}  \\
+ & {\delimit}  &{\terminal{ValueLE}} {\nonterminal{Value}} {\nonterminal{Value}}  \\
+ & {\delimit}  &{\terminal{ValueEQ}} {\nonterminal{Value}} {\nonterminal{Value}}  \\
+ & {\delimit}  &{\terminal{TrueObs}}  \\
+ & {\delimit}  &{\terminal{FalseObs}}  \\
+ & {\delimit}  &{\terminal{(}} {\nonterminal{Observation}} {\terminal{)}}  \\
+\end{tabular}\\
+
+\begin{tabular}{lll}
+{\nonterminal{Token}} & {\arrow}  &{\terminal{Token}} {\nonterminal{String}} {\nonterminal{String}}  \\
+ & {\delimit}  &{\terminal{(}} {\nonterminal{Token}} {\terminal{)}}  \\
+\end{tabular}\\
+
+\begin{tabular}{lll}
+{\nonterminal{Party}} & {\arrow}  &{\terminal{Address}} {\nonterminal{String}}  \\
+ & {\delimit}  &{\terminal{Role}} {\nonterminal{String}}  \\
+ & {\delimit}  &{\terminal{(}} {\nonterminal{Party}} {\terminal{)}}  \\
+\end{tabular}\\
+
+\begin{tabular}{lll}
+{\nonterminal{Payee}} & {\arrow}  &{\terminal{Account}} {\nonterminal{Party}}  \\
+ & {\delimit}  &{\terminal{Party}} {\nonterminal{Party}}  \\
+ & {\delimit}  &{\terminal{(}} {\nonterminal{Payee}} {\terminal{)}}  \\
+\end{tabular}\\
+
+\begin{tabular}{lll}
+{\nonterminal{ChoiceId}} & {\arrow}  &{\terminal{ChoiceId}} {\nonterminal{String}} {\nonterminal{Party}}  \\
+ & {\delimit}  &{\terminal{(}} {\nonterminal{ChoiceId}} {\terminal{)}}  \\
+\end{tabular}\\
+
+\begin{tabular}{lll}
+{\nonterminal{ValueId}} & {\arrow}  &{\nonterminal{String}}  \\
+\end{tabular}\\
+
+\begin{tabular}{lll}
+{\nonterminal{Bound}} & {\arrow}  &{\terminal{Bound}} {\nonterminal{Integer}} {\nonterminal{Integer}}  \\
+\end{tabular}\\
+
+\begin{tabular}{lll}
+{\nonterminal{ListBound}} & {\arrow}  &{\emptyP} \\
+ & {\delimit}  &{\nonterminal{Bound}}  \\
+ & {\delimit}  &{\nonterminal{Bound}} {\terminal{,}} {\nonterminal{ListBound}}  \\
+\end{tabular}\\
+
+\begin{tabular}{lll}
+{\nonterminal{Timeout}} & {\arrow}  &{\nonterminal{Integer}}  \\
+\end{tabular}\\
+
+
+
+\end{document}
+
```

Comment:
Are we checking this is generated correctly? I think if we are committing generated stuff to the repo we should check it is current with the CI. But I would rather not commit it. If we do commit unchecked generated stuff, maybe we should give it a date or a version number, cause it would effectively become an archive snapshot. (Next person that updates the source may or may not remember to update this)